### ISSUE TITLE
[Bug] Logging in after connecting telnyx causes LateInitialization error

### ISSUE THREAD

**Issue by icodelifee** [team-telnyx/flutter-voice-sdk#26]
**Bug Category**
- [x] Credential Login
- [ ] Token Login
- [ ] Local Audio Issue
- [ ] Remote Audio Issue
- [ ] Audio Device Switching
- [ ] Mute / Unmute
- [ ] Hold / Unhold
- [ ] Performance issues
- [ ] Other

**SDK Version**
telnyx_webrtc: latest

**Describe the bug**
No, it does not occur in the example app because the example does not log in right after you connect the client. I had this issue for a long time and the workaround I did to fix it was using a delay of 4 seconds to log in after you connect the client because the connect() method is not asynchronous, but in release the delay is not really working and users are not able to automatically login into the client.
![Screenshot 2022-10-26 at 4 52 44 PM](https://user-images.githubusercontent.com/18023153/198013963-cd66f22a-1b69-4a7c-8270-00daa1688d63.png)

**Expected behaviour**

**Logs**
Please provide any logs available to you, remember to enable verbose logging within the SDK.

**Comment by icodelifee** [comment]
logs: https://katb.in/imumumesagi

**Comment by stashtechnologies** [comment]
**LateInitializationError: Field '_socket@254383924' has not been initialized.**
same error

**Comment by icodelifee** [comment]
> Same Issue... still no solution

Did you try connecting after a delay?

**Comment by mohsinnaqvi606** [comment]
yes tried.. multiple time also tried web demo
https://webrtc.telnyx.com/

getting error   **"sipReason": "Connection has no Outbound Profile assigned D38",**

**Comment by mohsinnaqvi606** [comment]
**solved It**

this error comes when you try to connect client and login simultaneously. just connect client first and after some delay try to login. there will be no error

**Cheeers**

**Comment by icodelifee** [comment]
> **solved It**
> 
> this error comes when you try to connect client and login simultaneously. just connect client first and after some delay try to login. there will be no error
> 
> **Cheeers**

Its not really a solution, its a workaround

**Comment by mohsinnaqvi606** [comment]
you can not login until client is not connected & client it will take some time to connect.
so you have to wait for client connection

**Comment by icodelifee** [comment]
> you can not login until client is not connected & client it will take some time to connect. so you have to wait for client connection

Yes i know that, ive been doing that ever since i opened this issue, but you might still run into connection issues, if you have  a poor network or slow device.

**Comment by mohsinnaqvi606** [comment]
yes, it will be an error in that case. because there is no way to detect whether a connection is established on not.

**Comment by gada-raj** [comment]
Make the connect function Asynchronous so we can wait for it and then call the login function.

**Comment by Oliver-Zimmerman** [comment]
You will receive a message with Client Ready when you are connected. Only then can you login. 

@mohsinnaqvi606 @gada-raj
